operator/pkg/ingress/secretsync.go

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package ingress
+
+import (
+	"context"
+
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/cilium/cilium/pkg/logging/logfields"
+)
+
+func EnqueueReferencedTLSSecrets(c client.Client, logger logrus.FieldLogger) handler.EventHandler {
+	return handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, obj client.Object) []reconcile.Request {
+		scopedLog := logger.WithFields(logrus.Fields{
+			logfields.Controller: "secrets",
+			logfields.Resource:   obj.GetName(),
+		})
+
+		ing, ok := obj.(*networkingv1.Ingress)
+		if !ok {
+			return nil
+		}
+
+		// Check whether Ingress is managed by Cilium
+		if !isCiliumManagedIngress(ctx, c, *ing) {
+			return nil
+		}
+
+		var reqs []reconcile.Request
+		for _, tls := range ing.Spec.TLS {
+			s := types.NamespacedName{
+				Namespace: ing.Namespace,
+				Name:      string(tls.SecretName),
```

Comment:
could we add a check here for not queuing an empty secretname?

Reply:
good idea -> added